sql: fix ExtraColumnIDs not being populated for implicit indexes

otan opened the pull request:
We previously were missing ExtraColumnIDs being populated with implicit
partitioned columns on CREATE TABLE, as the logic to do this in
AllocateIDs occurs before the implicit partitioning columns are known.
This commit rectifies this.

Resolves #59583 

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59627)
<!-- Reviewable:end -->

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks!

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59627#-:-MSTI3B3DUu-nFY3tvdS:brfqlux)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/create_table.go, line 2037 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59627#-MSTIClL3tRlUpnVmKCb:-MSTIClL3tRlUpnVmKCc:bsq7k82) ([raw file](https://github.com/cockroachdb/cockroach/blob/a811cd1fabfafd9681254b84ef5cb2cdffa2fa33/pkg/sql/create_table.go#L2037)):*
> ```Go
> 			}
> 			// During CreatePartitioning, implicitly partitioned columns may be
> 			// created. This needs to happen before CreatePartitioning is called
> ```

[nit] what is meant by "This"? It sounds like it refers to the creation of implicitly partitioned columns, but the rest of the sentence seems to contradict that.


<!-- Sent from Reviewable.io -->

otan committed:
sql: fix ExtraColumnIDs not being populated for implicit indexes

We previously were missing ExtraColumnIDs being populated with implicit
partitioned columns on CREATE TABLE, as the logic to do this in
AllocateIDs occurs before the implicit partitioning columns are known.
This commit rectifies this.

Release note: None

otan commented:
fixed up!

bors r=rytaft
